dcos/nodeutil/util.go

```diff
@@ -0,0 +1,325 @@
+package nodeutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dcos/dcos-go/dcos"
+	"github.com/dcos/dcos-go/exec"
+)
+
+// ErrNodeUtil is an error structure raised by exported functions with meaningful error message.
+type ErrNodeUtil struct {
+	msg string
+}
+
+func (e ErrNodeUtil) Error() string {
+	return e.msg
+}
+
+// NodeInfo defines an interface to interact with DC/OS cluster via go methods.
+type NodeInfo interface {
+	DetectIP() (net.IP, error)
+	Role() (string, error)
+	IsLeader() (bool, error)
+	MesosID() (string, error)
+}
+
+// dcosInfo is implementation of NodeInfo interface.
+type dcosInfo struct {
+	sync.Mutex
```

Comment:
this is definitely great idea. However i'll be able to focus on this after December 1st. Then i'll have time for refactor / bug fixes.